example/storybook/stories/Notifications/index.tsx

```diff
@@ -0,0 +1,217 @@
+      body: 'Local notification!',
+      title: 'Local Notification Title',
+      sound: 'chime.aiff',
+      badge: 0,
+      type: '',
+      thread: '',
+      payload: {
+        category: 'SOME_CATEGORY',
+        link: 'localNotificationLink',
+      },
+      //@ts-ignore
+      android_channel_id: 'bma-example-channel',
+    });
+  };
+
+  const clearLocalNotifications = () => {
+    setEvents([]);
+  };
+
+  const renderOpenedNotification = (notification: Notification) => {
+    return (
+      <View style={styles.openedNotificationView}>
+        <Text style={styles.openedNotificationText}>Notification Opened</Text>
+        <Text>{`Title: ${notification.title}`}</Text>
+        <Text>{`Body: ${notification.body}`}</Text>
+      </View>
+    );
+  };
+
+  const renderReceivedNotification = (notification: Notification) => {
+    return (
+      <View style={styles.receivedNotificationView}>
+        <Text style={styles.recievedNotificationText}>
+          Notification Received
+        </Text>
+        <Text>{`Title: ${notification.title}`}</Text>
+        <Text>{`Body: ${notification.body}`}</Text>
+      </View>
+    );
+  };
+
+  const renderEvent = (event: Event) => {
+    if (event.type === 'notificationReceived') {
+      return renderReceivedNotification(event.notification);
+    }
+    return renderOpenedNotification(event.notification);
+  };
+
+  // Request the permissions to receive notifications
+  useEffect(() => {
+    requestNotificationsPermissions(({ deviceToken }) => {
+      if (deviceToken && account) {
+        setDeviceToken(deviceToken);
+
+        // Register the device with the LifeOmic platform to start receiving push notifications
+        registerDeviceToken({
+          deviceToken,
+          // TODO: update the application when the background allows BMAs
+          application: 'lifeResearch', // The application name will be provided by LifeOmic upon onboarding
+          httpClient,
+          accountId: account.id,
+        });
+      }
+    });
+  }, [account]);
+
+  // Set the notification channel for Android
+  useEffect(() => {
+    if (Platform.OS === 'android') {
+      Notifications.setNotificationChannel({
+        channelId: 'starter-example-channel',
+        name: 'Starter Example',
+        importance: 5,
+        description: 'Channel for the Starter App',
+        enableLights: true,
+        enableVibration: true,
+        showBadge: true,
+        vibrationPattern: [200, 1000, 500, 1000, 500],
+      });
+    }
+  }, []);
+
+  useEffect(() => {
+    // Handler called when a notification is pressed
+    onNotificationOpened((notification) => {
+      setEvents((events) => [
+        { type: 'notificationOpened', notification },
+        ...events,
+      ]);
+    });
+
+    onNotificationReceived((notification) => {
+      setEvents((events) => [
+        { type: 'notificationReceived', notification },
+        ...events,
+      ]);
+    });
+
+    const getInitial = async () => {
```

Comment:
@markdlv is this the initialization logic that you were talking (in a previous slack thread) about that should be either in the init file or as a context provider? in this case a provider since its got the setEvent hook call inside?